app/assets/javascripts/controllers/middleware_providers/_module.js

```diff
@@ -0,0 +1,50 @@
+miqHttpInject(angular.module('middleware.provider', ['miQStaticAssets', 'ui.bootstrap', 'ui.router', 'patternfly.select', 'ui.bootstrap.tabs', 'patternfly.views', 'ngAnimate']))
+.config(function($stateProvider, $locationProvider, $urlRouterProvider) {
+  $stateProvider.state('list_providers', {
+    url: '/ems_middleware/show_list',
+    templateUrl: '/static/middleware/list_providers.html',
+    controller: 'miqListProvidersController as vm'
```

Comment:
I though I answered, looks like I either forgot to hit comment.

I think many people have asked this question and I don't think we can find good answer to it, it's in a lot of angular tutorials and a lot of folks uses it this way. I myself looked for it couple of months back, why `controller as vm` is most used. Looks like someone somewhere used this and a lot of us (including me) followed. To be honest I kinda like this approach of two letters alias, otherwise you'd have to remember the name of controller (providerCrtl? pCtrl? or something along those lines).

We can have open discussion about it, I am opened to any suggestions (I worked with nameCrtl and more generic vm), it's completely up to us what we'll use. I used her the most generic `vm`, but if we'd prefer something more specific why not?

Also I sometimes explain **why** everybody uses `vm` as `ViewModel`.